track/models.py

```diff
@@ -1,18 +1,35 @@
 import io
 import datetime
 import csv
+import logging
 import typing
 from flask_pymongo import PyMongo
-from pymongo.errors import PyMongoError
+from pymongo.errors import PyMongoError, OperationFailure, ConnectionFailure
 import track.data
 
+
 # These functions are meant to be the only ones that access the g.db.db
 # directly. If we ever decide to migrate from tinyg.db.db, that can all be
 # coordinated here.
 
 db = PyMongo()
-
 QueryError = PyMongoError
+logger = logging.getLogger(__name__)
+
+def retry(num_tries, exceptions):
+    def decorator(func):
+        def f_retry(*args, **kwargs):
+            for i in range(num_tries):
+                try:
+                    return func(*args, **kwargs)
+                except exceptions as e:
+                    logging.exception("Exception hit in retry handler.")
```

Comment:
Tested and app behaved as before.  Just a small comment, to prevent confusion, should we just used one loggin object

logger.exception("Exception : {} hit in retry handler.".format(e))  , also, just using 'e' since you set it up :)

Reply:
"exception(msg, *args, **kwargs)
Logs a message with level ERROR on this logger. The arguments are interpreted as for debug(). Exception info is added to the logging message. This method should only be called from an exception handler."

No need for the format(e) since that's what .exception does... BUUUT I did have a bug in that the line should read logger.exception(   so thanks for the pickup. ;)

Reply:
ahh Ok :)  👍